streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -217,6 +220,21 @@ private ByteBuffer encodeVersionThree() {
         return buf;
     }
 
+    private ByteBuffer encodeVersionFour() {
+        final byte[] endPointBytes = prepareUserEndPoint();
+
+        final ByteBuffer buf = ByteBuffer.allocate(getVersionThreeByteLength(endPointBytes));
```

Comment:
add `getVersionFourByteLength()` ?

Or rename method to `getVersionThreeAndFourByteLength()` ?